### ISSUE TITLE
`slither-flat` doesn't flatten dYdX solo contracts correctly

### ISSUE THREAD

**Issue by Pet3ris** [crytic/slither#493]
Hi There,

Thanks for building `slither-flat`! It seems to be a lot more forgivign than `soljitsu` or `truffle-flatten`. I'm struggling, however, with using it to flatten dYdX contracts.

**How to reproduce**

```bash
#!/usr/bin/bash

# Dependency: npm, nvm
# Better not to run this as a script

# Get the repository
echo "solo" > .gitignore
git clone https://github.com/dydxprotocol/solo.git
cd solo

# Install javascript dependencies
npm i keccakjs --save
npm i solium@^1.0.0 --save
npm i acorn --save
npm i

# Testing that everything installed correctly (make sure Docker is enabled at this point)
nvm use 10.16.2  # Need this version to compile contracts
npm test

# Install and run slither-flat
cd ..
pipenv install slither-analyzer
pipenv run slither-flat solo
```

**Issues**

I noticed two issues with these contracts:

1) Duplicate contracts

In particular, `ReentrancyGuard` appeared many times in different contract files (and multiple times per file)

2) Missing characters

In particular, `ReentrancyGuard` appeared as `ontract ReentrancyGuard` (missing a `c`)

3) Some contracts don't exist in their own file anymore

Contract `external/OnlySolo.sol` doesn't have a corresponding `OnlySolo.sol` file in the output, instead it's embedded inline with other contracts in multiple places. [perhaps this is intentional]

**Potential problems**

With `truffle-flatten` it flagged a circular dependency in the contracts, so perhaps this is related to the issue here.

**Comment by montyly** [comment]
Hi @Pet3ris, thank you for reporting this.

We will fix it as soon as possible

**Comment by montyly** [comment]
Hi @Pet3ris, the issue is fixed with #494, and it is in master.

I tried on `dydxprotocol/solo.git` and all the flattened files generated can be compiled with solc 0.5.7. 

This will be included in the next release. If you want to try it before, I would recommend to use `virtualenv` and install slither from github. The procedure is described [here](https://github.com/crytic/slither/wiki/Developer-installation).

Don't hesitate to join our [slack](https://empireslacking.herokuapp.com/) (#ethereum) if you have any other questions

**Comment by Pet3ris** [comment]
Thanks a lot @montyly for the rapid fix! Can confirm the extra token issue got resolved for me. I'm guessing that the duplication of contracts within the files (1. and 3.) is intentional to have dependencies present internally in each contract.
